pkg/kubectl-argo-rollouts/info/rollout_info.go

```diff
@@ -122,10 +122,24 @@ func RolloutErrorConditions(ro *v1alpha1.Rollout) []string {
 	return errorConditions
 }
 
+// isGenerationObserved determines if the rollout spec has been observed by the controller. This
+// only applies to v0.10 rollout which uses a numeric status.observedGeneration. For v0.9 rollouts
+// and below this function always returns true.
+func isGenerationObserved(ro *v1alpha1.Rollout) bool {
+	observedGen, err := strconv.Atoi(ro.Status.ObservedGeneration)
+	if err != nil {
+		return true
+	}
+	return int64(observedGen) == ro.Generation
+}
+
 // RolloutStatusString returns a status and message for a rollout
 // This logic is more or less the same as the Argo CD rollouts health.lua check
 // Any changes to this function should also be changed there
 func RolloutStatusString(ro *v1alpha1.Rollout) (string, string) {
+	if !isGenerationObserved(ro) {
+		return "Progressing", "waiting for rollout spec update to be observed"
+	}
```

Comment:
This is the check that allows us to detect if rollout controller observed the new spec.